sql/core/src/test/scala/org/apache/spark/sql/execution/WholeStageCodegenSuite.scala

```diff
@@ -319,4 +319,15 @@ class WholeStageCodegenSuite extends QueryTest with SharedSQLContext {
       assert(df.limit(1).collect() === Array(Row("bat", 8.0)))
     }
   }
+
+  test("SPARK-25767: Lazy evaluated stream of expressions handled correctly") {
```

Comment:
Even without this patch this test would pass (I just tried it on master). A stream always evaluates its first element, so you probably need to add another key here.

Reply:
Strange, I've just rerun it on master and got:
```
08:34:45.754 ERROR org.apache.spark.sql.catalyst.expressions.codegen.CodeGenerator: failed to compile: org.codehaus.commons.compiler.CompileException: File 'generated.java', Line 81, Column 17: Expression "bhj_isNull_6" is not an rvalue
org.codehaus.commons.compiler.CompileException: File 'generated.java', Line 81, Column 17: Expression "bhj_isNull_6" is not an rvalue
```

Reply:
oh wait, I executed those commands in sparkShell and that falls back to interpreted mode when compilation fails. That is what happened here. Sorry about the fuss.